prov/sockets/src/sock_av.c

```diff
@@ -386,15 +386,16 @@ static int sock_av_insertsym(struct fid_av *av, const char *node, size_t nodecnt
 		fmt = 0;
 	else 
 		fmt = offset;
-
+	
+	assert((hostlen-offset) < FI_NAME_MAX);
 	strncpy(base_host, node, hostlen - (offset));
 	var_port = atoi(service);
 	var_host = atoi(node + hostlen - offset);
 	
 	for (i = 0; i < nodecnt; i++) {
 		for (j = 0; j < svccnt; j++) {
-			sprintf(tmp_host, "%s%0*d", base_host, fmt, var_host + i);
-			sprintf(tmp_port, "%d", var_port + j);
+			assert(snprintf(tmp_host, FI_NAME_MAX, "%s%0*d", base_host, fmt, var_host + i) < FI_NAME_MAX);
+			assert(snprintf(tmp_port, FI_NAME_MAX,  "%d", var_port + j) < FI_NAME_MAX);
```

Comment:
assert() will be compiled out in a non-debug build.  This will result in the snprintf being lost.